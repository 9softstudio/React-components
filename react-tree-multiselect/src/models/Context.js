import React from 'react';

export default React.createContext({
    rootNode: {},
    nodes: [],
    onToggle: () => { },
    onCheck: () => { }
});