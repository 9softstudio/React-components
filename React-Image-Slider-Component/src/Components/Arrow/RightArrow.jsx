import React, { Component } from 'react';
import Context from '../../slider-context';

export default class RightArrow extends Component {
    render() {
        return (
            <Context.Consumer>{
                ({ onRightArrowClick }) =>
                    <img src="img/slider-right-arrow.png" className="slider-arrow-position right-arrow" onClick={onRightArrowClick} />
            }</Context.Consumer>);
    }
};