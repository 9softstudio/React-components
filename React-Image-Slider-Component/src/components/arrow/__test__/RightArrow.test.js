import React from 'react';
import RightArrow from '../RightArrow';
import Context from '../../../slider-context';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const component = renderer
        .create(
            <Context.Provider value={({ onRightArrowClick: jest.fn() })}>
                <RightArrow />
            </Context.Provider>
        );
    expect(component.toJSON()).toMatchSnapshot();
});

test('on click: onRightArrowClick()', () => {
    const onRightArrowClick = jest.fn();
    const component = renderer
        .create(
            <Context.Provider value={({ onRightArrowClick: onRightArrowClick })}>
                <RightArrow />
            </Context.Provider>
        );
    expect(component.toJSON().props.onClick).toEqual(onRightArrowClick);
});