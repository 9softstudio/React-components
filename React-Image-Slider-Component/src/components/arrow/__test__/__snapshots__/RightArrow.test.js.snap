// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  className="slider-arrow-position right-arrow"
  onClick={[MockFunction]}
  src="img/slider-right-arrow.png"
/>
`;
